test/e2e_node/container_lifecycle_test.go

```diff
@@ -746,6 +750,161 @@ var _ = SIGDescribe(framework.WithNodeConformance(), "Containers Lifecycle", fun
 		framework.ExpectNoError(results.Starts(prefixedName(PreStopPrefix, regular1)))
 		framework.ExpectNoError(results.Exits(regular1))
 	})
+
+	ginkgo.When("a pod is terminating because its liveness probe fails", func() {
+		regular1 := "regular-1"
+
+		testPod := func() *v1.Pod {
+			return &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-pod",
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy:                 v1.RestartPolicyNever,
+					TerminationGracePeriodSeconds: ptr.To(int64(100)),
+					Containers: []v1.Container{
+						{
+							Name:  regular1,
+							Image: imageutils.GetE2EImage(imageutils.BusyBox),
+							Command: ExecCommand(regular1, execCommand{
+								Delay:              100,
+								TerminationSeconds: 15,
+								ExitCode:           0,
+							}),
+							LivenessProbe: &v1.Probe{
+								ProbeHandler: v1.ProbeHandler{
+									Exec: &v1.ExecAction{
+										Command: ExecCommand(prefixedName(LivenessPrefix, regular1), execCommand{
+											ExitCode:      1,
+											ContainerName: regular1,
+										}),
+									},
+								},
+								InitialDelaySeconds: 10,
+								PeriodSeconds:       1,
+								FailureThreshold:    1,
+							},
+						},
+					},
+				},
+			}
+		}
+
+		f.It("should execute readiness probe while in preStop, but not liveness", f.WithNodeConformance(), func() {
+			client := e2epod.NewPodClient(f)
+			podSpec := testPod()
+
+			ginkgo.By("creating a pod with a readiness probe and a preStop hook")
+			podSpec.Spec.Containers[0].Lifecycle = &v1.Lifecycle{
+				PreStop: &v1.LifecycleHandler{
+					Exec: &v1.ExecAction{
+						Command: ExecCommand(prefixedName(PreStopPrefix, regular1), execCommand{
+							Delay:         1,
+							ExitCode:      0,
+							ContainerName: regular1,
+						}),
+					},
+				},
+			}
+			podSpec.Spec.Containers[0].ReadinessProbe = &v1.Probe{
+				ProbeHandler: v1.ProbeHandler{
+					Exec: &v1.ExecAction{
+						Command: ExecCommand(prefixedName(ReadinessPrefix, regular1), execCommand{
+							ExitCode:      0,
+							ContainerName: regular1,
+						}),
+					},
+				},
+				InitialDelaySeconds: 1,
+				PeriodSeconds:       1,
+			}
+
+			preparePod(podSpec)
+
+			podSpec = client.Create(context.TODO(), podSpec)
+
+			ginkgo.By("Waiting for the pod to complete")
+			err := e2epod.WaitForPodNoLongerRunningInNamespace(context.TODO(), f.ClientSet, podSpec.Name, podSpec.Namespace)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("Parsing results")
+			podSpec, err = client.Get(context.TODO(), podSpec.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+			results := parseOutput(context.TODO(), f, podSpec)
+
+			ginkgo.By("Analyzing results")
+			// readiness probes are called during pod termination
+			framework.ExpectNoError(results.RunTogether(prefixedName(PreStopPrefix, regular1), prefixedName(ReadinessPrefix, regular1)))
```

Comment:
@matthyx 
I think we have to give more than 1 second for the preStop hook to deflake [this](https://github.com/kubernetes/kubernetes/issues/124791#issue-2289457502).

(Not sure what's wrong with the OOM case though...)